src/metadata.h

```diff
@@ -687,6 +723,62 @@ namespace librealsense
     };
     REGISTER_MD_TYPE(md_safety_info, md_type::META_DATA_INTEL_SAFETY_ID)
 
+    struct md_mapping_common_info
+    {
+        uint32_t    frame_counter;
+        uint32_t    depth_frame_counter;
+        uint32_t    frame_timestamp;
+        uint32_t    floor_plane_equation_a;
+        uint32_t    floor_plane_equation_b;
+        uint32_t    floor_plane_equation_c;
+        uint32_t    floor_plane_equation_d;
+        uint8_t     safety_preset_id;
+    };
+
+    /**\brief md_occupancy - Occupancy Frame */
+    struct md_occupancy
+    {
+        md_header   header;
+        uint32_t    version;
+        uint32_t    flags;
+        uint32_t    frame_counter;
+        uint32_t    depth_frame_counter;
+        uint32_t    frame_timestamp;
+        uint32_t    floor_plane_equation_a;
+        uint32_t    floor_plane_equation_b;
+        uint32_t    floor_plane_equation_c;
+        uint32_t    floor_plane_equation_d;
+        uint8_t     safety_preset_id;
+        uint16_t    grid_rows;
+        uint16_t    grid_columns;
+        uint8_t     cell_size;
+        uint8_t     reserved[32];
+        uint16_t    padding;
+        uint32_t    payload_crc32;
+    };
+    REGISTER_MD_TYPE(md_occupancy, md_type::META_DATA_INTEL_OCCUPANCY_ID)
+
+    /**\brief md_point_cloud - Point Cloud Frame */
+    struct md_point_cloud
+    {
+        md_header   header;
+        uint32_t    version;
```

Comment:
And here?

Reply:
Not possible either - our structs need to be flat